File: app/schemas/knowledge.py
import uuid
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel

class KnowledgeBaseBase(BaseModel):
    name: str
    description: Optional[str] = None
    type: str = "text"  # text, excel, or pageindex
    group_id: Optional[uuid.UUID] = None

class KnowledgeBaseCreate(KnowledgeBaseBase):
    pass

class KnowledgeBaseUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    type: Optional[str] = None
    group_id: Optional[uuid.UUID] = None

class KnowledgeBaseGroupBase(BaseModel):
    name: str
    description: Optional[str] = None

class KnowledgeBaseGroupCreate(KnowledgeBaseGroupBase):
    pass

class KnowledgeBaseGroupUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None

class KnowledgeBaseGroupResponse(KnowledgeBaseGroupBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

class DocumentResponse(BaseModel):
    id: uuid.UUID
    knowledge_base_id: uuid.UUID
    filename: str
    file_type: str
    status: str
    error_message: Optional[str]
    chunk_count: int
    extra_metadata: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime

class KnowledgeBaseResponse(KnowledgeBaseBase):
    id: uuid.UUID
    is_published: bool
    group_id: Optional[uuid.UUID] = None
    created_at: datetime
    updated_at: datetime
    documents: List[DocumentResponse] = []

class KnowledgeBaseListResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: Optional[str]
    type: str
    is_published: bool
    document_count: int
    group_id: Optional[uuid.UUID]
    group_name: Optional[str] = None
    created_at: datetime
    updated_at: datetime

class SearchRequest(BaseModel):
    query: str
    top_k: int = 5
    score_threshold: float = 0.0

class SearchResult(BaseModel):
    id: str
    content: str
    metadata: Dict[str, Any]
    score: float

class PageIndexNode(BaseModel):
    title: str
    start_index: int
    end_index: int
    node_id: str
    summary: str

class PageIndexStructure(BaseModel):
    doc_name: str
    structure: List[PageIndexNode]

class PageIndexSearchResult(BaseModel):
    node: PageIndexNode
    page_content: str
    score: float

class PageIndexSearchResponse(BaseModel):
    results: List[PageIndexSearchResult]
    prompt: Optional[str] = None

class SearchResponse(BaseModel):
    results: List[SearchResult]

class BatchDeleteRequest(BaseModel):
    ids: List[uuid.UUID]

File: frontend/src/types/knowledge.ts
export interface Document {
  id: string;
  knowledge_base_id: string;
  filename: string;
  file_type: string;
  status: 'pending' | 'processing' | 'completed' | 'error';
  error_message?: string;
  chunk_count: number;
  extra_metadata?: {
    excel_columns?: string[];
  };
  created_at: string;
  updated_at: string;
}

export interface KnowledgeBaseGroup {
  id: string;
  name: string;
  description?: string;
  created_at: string;
  updated_at: string;
}

export interface KnowledgeBase {
  id: string;
  name: string;
  description?: string;
  type: 'text' | 'excel' | 'pageindex';
  is_published: boolean;
  group_id?: string;
  group_name?: string;
  created_at: string;
  updated_at: string;
  document_count?: number;
  documents?: Document[];
}

export interface KnowledgeBaseCreate {
  name: string;
  description?: string;
  type?: 'text' | 'excel' | 'pageindex';
  group_id?: string;
}

export interface SearchResult {
  id: string;
  content: string;
  metadata: Record<string, any>;
  score: number;
}

export interface PageIndexNode {
  title: string;
  start_index: number;
  end_index: number;
  node_id: string;
  summary: string;
}

export interface PageIndexSearchResult {
  node: PageIndexNode;
  page_content: string;
  score: number;
}

export interface PageIndexSearchResponse {
  results: PageIndexSearchResult[];
  prompt?: string;
}
